src/actions/commands/actions.ts

```diff
@@ -653,13 +653,66 @@ class CommandMoveHalfPageDown extends CommandEditorScroll {
   keys = ['<C-d>'];
   to: EditorScrollDirection = 'down';
   by: EditorScrollByUnit = 'halfPage';
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    let lineOffset = 0;
```

Comment:
I think putting this in the base class `CommandEditorScroll` is a better place for this?

Reply:
I'll take a deeper look into how Ctrl+F and others work and then move it into them. I have a feeling that they don't work properly.